### ISSUE TITLE
[BUG] PR 65989 broke stuff in my environment

### ISSUE THREAD

**Issue by darkpixel** [saltstack/salt#66121]
**Description**

I upgraded my salt master to 3006.7 and a bunch of 3006.6 minions stopped communicating with the server with key errors.

I believe this is related to https://github.com/saltstack/salt/pull/65989

On these boxes the various minion pki files were all generated (and get re-generated) with an EOL at the end of ```minion.pub```.

**Steps to Reproduce the behavior**
Find a box where minion.pub has an EOL, or just add it with your favorite text editor.
Stop the salt-minion process.
Delete the key from the master.
Delete the keys from the ```salt/pki/minion``` folder.
Run salt-call -l info test.ping.
You'll get a message that the key needs to be accepted, go accept it.
Run salt-call -l info test.ping again.
The key will be invalid and a new key will show up as Rejected on the server.

vi -b /etc/salt/pki/minion/minion.pub
Issue the ```:set noeol``` command in vi.
Save and exit.
Run salt-call -l info test.ping again and it will succeed.

**Comment by Z10N0110** [comment]
I just got the similar situation here, some new installed 3006.7 minions cannot connect to 3006.4 master.

**Comment by mdschmitt** [comment]
Never ever ever upgrade minions before your masters.

**Comment by mdschmitt** [comment]
The bug behavior described in the OP is valid/correct but running newer minions than masters is not supported and probably never will be.

**Comment by Z10N0110** [comment]
@mdschmitt Even the same major version will have incompatible problems? That's ... odd.

Those 3006.7 minions were installed via the official bootstrap script with args `-PX onedir 3006` on  Ubuntu 22.04 Linux node. After taking a closer look at the script, I found that apt repository file that script generated just use the `3006` path which will be only stores the latest `3006` version packages.

I just fixed it by do the version locking to `3006.4`, now the args is `-PX onedir 3006.4`.
